devbox.go

```diff
@@ -137,20 +137,22 @@ func (d *Devbox) Build(flags *docker.BuildFlags) error {
 
 // Plan creates a plan of the actions that devbox will take to generate its
 // shell environment.
-func (d *Devbox) ShellPlan() *plansdk.Plan {
-	// TODO: Move shell plan to a separate struct from build plan.
-	return d.convertToPlan()
+func (d *Devbox) ShellPlan() *plansdk.ShellPlan {
+	shellPlan := planner.GetShellPlan(d.srcDir, d.cfg.Packages)
+	shellPlan.DevPackages = d.cfg.Packages
```

Comment:
Since shell plans are only selected when users have all of the packages a shell plan expect, user defined packages will always include `devPackages` from shell plan.

I was thinking `planner.GetShellPlan` only strictly return the shell plan from planners.